app/namespace-watcher.go

```diff
@@ -2,39 +2,64 @@ package main
 
 import (
 	"context"
-	"fmt"
-	"log"
 	"os"
 	"strings"
 
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/watch"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 )
 
-func createLimitRange(clientset *kubernetes.Clientset, namespaceName string) error {
+const version = "v1.0-alpha9"
 
-	version := "v1.0-alpha"
+type Limits struct {
+	CpuLimitMax              string
+	MemLimitMax              string
+	EphemeralStorageLimitMax string
+	CpuLimitMin              string
+	MemLimitMin              string
+	EphemeralStorageLimitMin string
+}
+
+var setLimits Limits
+
+func createOrUpdateLimitRange(clientset *kubernetes.Clientset, namespaceName string, limits Limits) error {
+
+	cpuLimitMin, err := resource.ParseQuantity(limits.CpuLimitMin)
+	if err != nil {
+		logrus.Fatalf("error parsing CPU_LIMIT_MIN: ", err)
```

Comment:
It should return the error, given this function returns a `error` type. The `logrus.Fatalf` would panic